demo-simple/src/main/java/com/novoda/downloadmanager/demo/MainActivity.java

```diff
@@ -123,14 +122,14 @@ protected void onCreate(Bundle savedInstanceState) {
 
     private final View.OnClickListener downloadBatchesOnClick = v -> {
         Batch batch = new Batch.Builder(BATCH_ID_1, "Made in chelsea")
-                .addFile(FILE_ID_1, FIVE_MB_FILE_URL, "foo/bar/5mb.zip")
-                .addFile(TEN_MB_FILE_URL)
+                .addFile(FIVE_MB_FILE_URL).withDownloadFileId(FILE_ID_1).withRelativePath("foo/bar/5mb.zip").apply()
+                .addFile(TEN_MB_FILE_URL).apply()
```

Comment:
The `apply` returns the parent builder (`Batch.Builder`) so the user can continue to add additional files and build the final batch.